apps/backend/src/datasources/postgres/StatusActionsDataSource.ts

```diff
@@ -0,0 +1,228 @@
+import { logger } from '@user-office-software/duo-logger';
+import { GraphQLError } from 'graphql';
+
+import {
+  ConnectionHasStatusAction,
+  ProposalStatusAction,
+  ProposalStatusActionType,
+} from '../../models/ProposalStatusAction';
+import { Rejection } from '../../models/Rejection';
+import { AddConnectionStatusActionsInput } from '../../resolvers/mutations/settings/AddConnectionStatusActionsMutation';
+import {
+  EmailActionConfig,
+  ProposalStatusActionConfig,
+  RabbitMQActionConfig,
+} from '../../resolvers/types/ProposalStatusActionConfig';
+import { StatusActionsDataSource } from '../StatusActionsDataSource';
+import database from './database';
+import {
+  ProposalStatusActionRecord,
+  ProposalWorkflowConnectionHasActionsRecord,
+} from './records';
+
+export default class PostgresStatusActionsDataSource
+  implements StatusActionsDataSource
+{
+  private createStatusActionConfig(
+    type: ProposalStatusActionType,
+    config: typeof ProposalStatusActionConfig
+  ) {
+    switch (type) {
+      case ProposalStatusActionType.EMAIL: {
+        const blankConfig = new EmailActionConfig();
+
+        return Object.assign(blankConfig, config);
+      }
+      case ProposalStatusActionType.RABBITMQ: {
+        const blankConfig = new RabbitMQActionConfig();
+
+        return Object.assign(blankConfig, config);
+      }
+      default:
+        return null;
+    }
+  }
+
+  private createConnectionStatusActionObject(
+    proposalActionStatusRecord: ProposalWorkflowConnectionHasActionsRecord & {
+      proposal_status_action_id: number;
+      name: string;
+      type: ProposalStatusActionType;
+      config: typeof ProposalStatusActionConfig;
+    }
+  ) {
+    return new ConnectionHasStatusAction(
+      proposalActionStatusRecord.connection_id,
+      proposalActionStatusRecord.proposal_status_action_id,
+      proposalActionStatusRecord.workflow_id,
+      proposalActionStatusRecord.name,
+      proposalActionStatusRecord.type,
+      proposalActionStatusRecord.executed,
+      this.createStatusActionConfig(
+        proposalActionStatusRecord.type,
+        proposalActionStatusRecord.config
+      )
+    );
+  }
+
+  private createProposalStatusActionObject(
+    proposalActionStatusRecord: ProposalStatusActionRecord
+  ) {
+    return new ProposalStatusAction(
+      proposalActionStatusRecord.proposal_status_action_id,
+      proposalActionStatusRecord.name,
+      proposalActionStatusRecord.description,
+      proposalActionStatusRecord.type
+    );
+  }
+
+  async getConnectionStatusActions(

```

Comment:
Should the stuff moved here be removed from `datasources/postgres/ProposalSettingsDataSource.ts`?

Reply:
That definitely needs to be removed. Nice catch!